src/DicomMetaDictionary.js

```diff
@@ -126,6 +126,10 @@ class DicomMetaDictionary {
                     // when the vr is data-dependent, keep track of the original type
                     naturalDataset._vrMap[naturalName] = data.vr;
                 }
+
+                if (data.vr == "UN") {
```

Comment:
I agree with @pieper that this should be a separate step that should be configurable - maybe as an option going into the parse step, but also as a stand alone filter that works on any JSON DICOMweb metadata instance.  That filter should then have some way to register named fixes and a way to disable fixes individually.

Reply:
Could you please clarify this point, I want to make sure that I understood you correctly. Do you mean that  I should add an option(flag) to `naturalizeDataset/denaturalizeDataset` and depending on this option(flag) apply or skip for this logic?

Reply:
I was thinking you could have a special-purpose piece of code at the application level that is coded based on the unique information that it will be accepting data from a source (kpacs) that has known limitations in terms of generating standard dicom objects.  I.e. in this case you know that certain tag values will have incorrect values.  This special-purpose code should read the instance and apply whatever fixes are needed to make the instance standards-compliant and then send it to the dcmjs code.

We may want to provide a standard place for this kind of special-purpose code to happen.  Perhaps in dcmjs-dimse there could be a hook to apply a standardizing transform to the incoming instance data.  If there end up being very common ones they could be bundled as add-on packages (e.g. a set of transforms for kpacs, ones for non-standard microCT scanners, etc).

Reply:
We don't have access to this low-level data on the application level as the application provides C-Store service only thing we have from dcmjs-dimse is Dataset without any VR information. dcmjs-dimse relies on dcmjs while reading the Dataset. In this case, we don't know if something is wrong until we have read the data.
What you propose is rewriting parts of dcmjs and dcmjs-dimse that are receiving binary data over the network and naturalizing the dataset. 
The other thing is that we don't have access to the KPacs dictionary and can't compare it to dcmjs Dictionary to find out which tags can have the wrong VRs.

Reply:
From what I understand if dcmjs-dimse is processing a c-store you should be able to access the instance as it is received.  From there you can read it as pure dicom json model and you can look at each of the tags in the instance and see if they have the non-standard form that leads to the issue based on the dicom dictionary and your knowledge of the deficiencies with kpacs.  (That is, if you see that a tag with a know standard VR has been set to UN by kpacs).  If you see an issue, you can modify the instances and save them back before further processing.

Reply:
At the application level in CStore we do not have DICOM json, we only have naturalized Dataset. At this level, there is no way how we could get the original VR for tag.
![image](https://github.com/dcmjs-org/dcmjs/assets/58589743/f4f43fa3-4a48-4df9-97e2-0552c433ce87)

DICOM json object is present inside dsmjs-dmse in the buffer/file reading functions.
![image](https://github.com/dcmjs-org/dcmjs/assets/58589743/0c4760c0-fb59-4d83-ad0f-e3b38bcebef8)
![image](https://github.com/dcmjs-org/dcmjs/assets/58589743/ffe1e1ac-a231-4925-ad87-38df0ea6821f)

If we talk about saving the original UN VR and saving it to a file without getting an error (as suggested in PR). I think it’s impossible to do without changes in dcmjs. The denaturalizeDataset takes the VR based only on the value in the dictionary, which leads to a mismatch between the VR and the value. If we talk about converting UN VR value to VR from the dictionary, it looks like a more difficult task to handle all possible VRs. @PantelisGeorgiadis do you have any ideas how this issue could be solved at the dcmjs-dimse level?